src/MediatR/Mediator.cs

```diff
@@ -48,19 +48,22 @@ public Task<TResponse> Send<TResponse>(IRequest<TResponse> request, Cancellation
                 throw new ArgumentNullException(nameof(request));
             }
             var requestType = request.GetType();
-            var requestInterfaceType = requestType
-                .GetInterfaces()
-                .FirstOrDefault(i => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(IRequest<>));
-            var isValidRequest = requestInterfaceType != null;
-
-            if (!isValidRequest)
-            {
-                throw new ArgumentException($"{nameof(request)} does not implement ${nameof(IRequest)}");
-            }
-
-            var responseType = requestInterfaceType!.GetGenericArguments()[0];
             var handler = _requestHandlers.GetOrAdd(requestType,
-                t => Activator.CreateInstance(typeof(RequestHandlerWrapperImpl<,>).MakeGenericType(requestType, responseType)));
+                t =>
+                {
+                    var requestInterfaceType = requestType

```

Comment:
Should this use `t` to avoid allocations?

Reply:
good catch. fixed